backend/src/main.rs

```diff
@@ -1,6 +1,60 @@
 #![recursion_limit = "256"]
-#![deny(clippy::unwrap_used, clippy::expect_used)]
+// Enable all lints apart from clippy::restriction by default.
+// See https://rust-lang.github.io/rust-clippy/master/index.html#blanket_clippy_restriction_lints for as to why restriction is not enabled.
+#![warn(clippy::pedantic)]
+#![warn(clippy::nursery)]
+#![warn(clippy::cargo)]
+// Lints in clippy::restriction which seem useful.
+#![warn(
+    clippy::clone_on_ref_ptr,
+    clippy::empty_structs_with_brackets,
+    clippy::exit,
+    clippy::expect_used,
+    clippy::format_push_string,
+    clippy::get_unwrap,
+    clippy::if_then_some_else_none,
+    clippy::integer_division,
+    clippy::large_include_file,
+    clippy::missing_docs_in_private_items,
+    clippy::mixed_read_write_in_expression,
+    clippy::multiple_inherent_impl,
+    clippy::mutex_atomic,
+    clippy::panic_in_result_fn,
+    clippy::partial_pub_fields,
+    clippy::print_stderr,
+    clippy::print_stdout,
+    clippy::rc_buffer,
+    clippy::rc_mutex,
+    clippy::rest_pat_in_fully_bound_structs,
+    clippy::same_name_method,
+    clippy::shadow_unrelated,
+    clippy::str_to_string,
+    clippy::string_to_string,
+    clippy::suspicious_xor_used_as_pow,
+    clippy::todo,
+    clippy::try_err,
+    clippy::unimplemented,
+    clippy::unnecessary_self_imports,
+    clippy::unneeded_field_pattern,
+    clippy::unreachable,
+    clippy::unseparated_literal_suffix,
+    clippy::unwrap_in_result,
+    clippy::unwrap_used,
+    clippy::use_debug,
+    clippy::verbose_file_reads
+)]
+// Cannot fix some errors because dependecies import them.
```

Comment:
Please write which ones are effected (so that we might reenable if the dep is gone).

Reply:
In this specific case it is `time: 0.1.45, 0.3.17` and `wasi: 0.10.0+wasi-snapshot-preview1, 0.11.0+wasi-snapshot-preview1`. Both of them are sub-dependencies of other crates. So I would keep `#![allow(clippy::multiple_crate_versions)]` as we would never really know what might happen when we add new crates.

Reply:
Actually I think that in such cases most likely the crates should update their dependencies. So if multiple_crate_versions would be used by many applications, libs depending on old libs would probably happen less. But of course we cannot influence it directly and during dev it would be rather disturbing, so let us close this topic for now.